runelite-client/src/main/java/net/runelite/client/plugins/tzhaartimers/TzhaarTimersPlugin.java

```diff
@@ -0,0 +1,297 @@
+				{
+					if (lastTime != null)
+					{
+						startTime = startTime.plus(Duration.between(startTime, now)).minus(Duration.between(startTime, lastTime));
+						lastTime = null;
+					}
+					if (checkInFightCaves())
+					{
+						infoBoxManager.removeInfoBox(timer);
+						createTimer(FIRE_CAPE, startTime, lastTime);
+					}
+					if (checkInInferno())
+					{
+						infoBoxManager.removeInfoBox(timer);
+						createTimer(INFERNAL_CAPE, startTime, lastTime);
+					}
+				}
+				if (message.contains(FIGHT_CAVE_PAUSED_MESSAGE) || message.contains(INFERNO_PAUSED_MESSAGE))
+				{
+					if (checkInFightCaves())
+					{
+						infoBoxManager.removeInfoBox(timer);
+						createTimer(FIRE_CAPE, startTime, now);
+					}
+					if (checkInInferno())
+					{
+						infoBoxManager.removeInfoBox(timer);
+						createTimer(INFERNAL_CAPE, startTime, now);
+					}
+					lastTime = now;
+				}
+				if (message.contains(DEFEATED_MESSAGE) || message.contains(INFERNO_COMPLETE_MESSAGE) || message.contains(FIGHT_CAVES_COMPLETE_MESSAGE))
+				{
+					removeTimer();
+					resetConfig();
+					startTime = null;
+					lastTime = null;
+					started = false;
+				}
+			}
+		}
+	}
+
+	private void updateInfoBoxState()
+	{
+		if (timer == null)
+		{
+			return;
+		}
+
+		if ((!checkInFightCaves() && !checkInInferno()) || !config.tzhaarTimers())
+		{
+			removeTimer();
+			resetConfig();
+			startTime = null;
+			lastTime = null;
+			started = false;
+		}
+	}
+
+	private boolean checkInFightCaves()
+	{
+		return client.getMapRegions() != null && Arrays.stream(client.getMapRegions())
+				.filter(x -> x == 9551)
+				.toArray().length > 0;
+	}
+
+	private boolean checkInInferno()
+	{
+		return client.getMapRegions() != null && Arrays.stream(client.getMapRegions())
+				.filter(x -> x == 9043)
+				.toArray().length > 0;
+	}
+
+	private void removeTimer()
+	{
+		infoBoxManager.removeInfoBox(timer);
+		timer = null;
+	}
+
+	private void createTimer(int id, Instant time, Instant lTime)
+	{
+		timer = new TzhaarTimers(itemManager.getImage(id), this, time, lTime);
+		infoBoxManager.addInfoBox(timer);
+	}
+
+	private void loadConfig()
+	{
+		startTime = configManager.getConfiguration(TzhaarTimersConfig.CONFIG_GROUP, TzhaarTimersConfig.CONFIG_TIME, Instant.class);
+		started = configManager.getConfiguration(TzhaarTimersConfig.CONFIG_GROUP, TzhaarTimersConfig.CONFIG_STARTED, Boolean.class);
+		lastTime = configManager.getConfiguration(TzhaarTimersConfig.CONFIG_GROUP, TzhaarTimersConfig.CONFIG_LASTTIME, Instant.class);
+	}
+
+	private void resetConfig()
+	{
+		configManager.unsetConfiguration(TzhaarTimersConfig.CONFIG_GROUP, TzhaarTimersConfig.CONFIG_TIME);
+		configManager.unsetConfiguration(TzhaarTimersConfig.CONFIG_GROUP, TzhaarTimersConfig.CONFIG_STARTED);
+		configManager.unsetConfiguration(TzhaarTimersConfig.CONFIG_GROUP, TzhaarTimersConfig.CONFIG_LASTTIME);
+	}
+
+	private void saveConfig()
+	{
+		if (startTime != null)
+		{
+			resetConfig();
```

Comment:
I don't think this is needed here because you are overwriting the config already anyway